R/SupervisedTask.R

```diff
@@ -1,13 +1,13 @@
-makeSupervisedTask = function(type, data, target, weights, blocking, fixup.data, check.data) {
-  task = makeTask(type = type, data = data, weights = weights, blocking = blocking, fixup.data = fixup.data, check.data = check.data)
+makeSupervisedTask = function(type, data, target, weights, blocking, fixup.data, check.data, spatial) {
+  task = makeTask(type = type, data = data, weights = weights, blocking = blocking, fixup.data = fixup.data, check.data = check.data, spatial = spatial)
 
   if (check.data) {
     # costsens does not have a target col...
     # assertCharacter(target, any.missing = FALSE, min.len = 1L)
     w = which.first(target %nin% colnames(data))
     if (length(w) > 0L)
       stopf("Column names of data doesn't contain target var: %s", target[w])
-    checkTaskData(task$env$data, cols = setdiff(colnames(data), target))
+    checkTaskData(task$env$data, cols = setdiff(colnames(task$env$data), target))
```

Comment:
@larskotthoff @berndbischl  pls comment on this. I still favour the implementation as it is right now. Also changing it would probably cost me couple of days to get it running as smooth as it is now. However, I would like to see it merged into `mlr` within the next weeks. We have already couple of people here using my branched version right now for a paper and it would be great if we could reference to `mlr` master. So if you insist on changing it, I will do it. I will also write a tutorial page for it after this is done here.